quickjs.c

```diff
@@ -39964,6 +39964,37 @@ static JSValue js_iterator_proto_iterator(JSContext *ctx, JSValue this_val,
     return js_dup(this_val);
 }
 
+static JSValue js_iterator_proto_get_toStringTag(JSContext *ctx, JSValue this_val)
+{
+    return JS_AtomToString(ctx, JS_ATOM_Iterator);
+}
+
+static JSValue js_iterator_proto_set_toStringTag(JSContext *ctx, JSValue this_val, JSValue val)
+{
+    JSObject *p;
+    int res;
+
+    if (JS_VALUE_GET_TAG(this_val) != JS_TAG_OBJECT)
+        return JS_ThrowTypeError(ctx,
+                                 "set Iterator.prototype[Symbol.toStringTag] called on non-object");
+    p = JS_VALUE_GET_OBJ(this_val);
+    if (p->class_id == JS_CLASS_ITERATOR) {
+        // This is home.
+        return JS_ThrowTypeError(ctx, "Cannot assign to read only property");
+    }
+    res = JS_GetOwnProperty(ctx, NULL, this_val, JS_ATOM_Symbol_toStringTag);
```

Comment:
For my curiosity: the JS_GetOwnProperty is because it's observable?

Reply:
Yes, AFAICT. The spec says to get the descriptor and either set or define a new property based on its presence.